lib/travis/build/errors.rb

```diff
@@ -32,6 +32,16 @@ def doc_path
       end
     end
 
+    class DeployConditionError < DeployConfigError
+      def initialize(msg = "The \\`deploy.on\\` should be a hash (dictionary).")
```

Comment:
Maybe add *key* i.e.

`"The \\`deploy.on\\` key should be a hash (dictionary).`

Reply:
This form is consistent with other error messages that we have. Also, strictly speaking, it is the _value_ that should be a hash, not the key itself (which is `on`)

Reply:
How about removing `The` ?

Or "The `deploy.on` *value*..."